internal/dispatch/caching/caching.go

```diff
@@ -183,7 +188,24 @@ func (cd *Dispatcher) SetDelegate(delegate dispatch.Dispatcher) {
 // DispatchCheck implements dispatch.Check interface
 func (cd *Dispatcher) DispatchCheck(ctx context.Context, req *v1.DispatchCheckRequest) (*v1.DispatchCheckResponse, error) {
 	cd.checkTotalCounter.Inc()
+
+	// Load the relation to get its computed cache key, if any.
+	// NOTE: We do not use the canonicalized cache key when checking within the same namespace, as
+	// we may get different results if the subject being checked matches the resource exactly.
 	requestKey := dispatch.CheckRequestToKey(req)
+	if cd.nsm != nil && req.ObjectAndRelation.Namespace != req.Subject.Namespace {
```

Comment:
I suppose, although the only reason we have this is to remove the need for a mock namespace manager in the caching testing. Do you prefer I just mock out the nsm there?

Reply:
Can we not just use a real namespace manager in the tests?

Reply:
Maybe there's value to a "KeyProvider" abstraction, and the disptacher just takes a reference to a key provider.

When there's a datastore available, we use a key provider backed w/ a nsm so that we get the pre-calculated cache keys, but for testing we can inject a simple static keyer, etc

Reply:
A good idea